pkg/cluster/install.go

```diff
@@ -288,23 +288,39 @@ func setFieldCreatedByHive(createdByHive bool) database.OpenShiftClusterDocument
 }
 
 func (m *manager) bootstrap() []steps.Step {
-	s := []steps.Step{
-		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.validateResources),
+	s := []steps.Step{}
+
+	if m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		s = append(s,
+			steps.Action(m.ensureClusterMsiCertificate),
+			steps.Action(m.initializeClusterMsiClients),
+		)
+	}
```

Comment:
Not entirely related to your PR, but since you're introducing `if` blocks during bootstrap, should `createOIDC` be included in one as well?

I know order matters, but it seems like we have some wi/mi steps in if blocks now, and others in line with the rest.

Reply:
It looks like the reason that `createOIDC` doesn't need to be in an `if` statement is because it does nothing and returns `nil` when called for a CSP cluster, which I agree is a bit confusing now that you point it out. I think refactoring that (if we want to) would be out of scope for this PR though.

Reply:
To be clear, when I say "refactor", I mean:

1. Modify `createOIDC` such that it returns an error when called on a CSP cluster
2. Update the bootstrap steps such that `createOIDC` only gets called for MIWI clusters

If I were to decide how to refactor it myself, this would make the most sense to me, because this way if we accidentally call a MIWI step on a CSP cluster as we're developing locally, we get an error that informs us we shouldn't have called that step.

Reply:
With that in mind, I think I'll update the two MIWI steps I've added in this PR to error out if called on CSP clusters.

Reply:
On second thought, I think I'm overcomplicating things and I should just leave my new code how it is. Let me know if you have other thoughts though.